### DISCUSSION TITLE
I'm trying to compile fabric 18.x and getting an error

### DISCUSSION THREAD

**Discussion by HydraheadHunter** [maruohon/malilib#83] [General]
I'm following the instructions to compile, but am getting an error. At first it was a refusing compile because I had the wrong java version installed, but now that that's fixed,  it's starting to compile but giving the following error:

> Task :remapJar FAILED
FAILURE: Build failed with an exception 
>* What went wrong:
>Execution failed for task ':remapJar'.
> A failure occurred while executing net.fabricmc.loom.task.RemapJarTask$RemapAction
   > Failed to remap

I'm getting this error on the fabric 18.x version's direct download without having changed anything yet. Is there a specific reason I'd be running into this error?

**Comment by maruohon** [comment]
Hm, no idea why it wouldn't work... I don't really feel like trying to nuke my environment to try with a fresh download.

But in my case I have Java 17 JDK installed from the Temurin builds (old AdoptOpenJDK) on Linux, and if I invoke the build with:
```
JAVA_HOME=/usr/lib/jvm/java-17-temurin/ bash gradlew build
```
it builds the mod succesfully.

Which Loom version does it say it's using for you? For me it seems to use Loom 0.11.34 currently.

**Reply by HydraheadHunter** [reply]
It says it's "Fabric Loom: 0.11.34," same as you. I'm on windows because I have a deep and rational fear of getting into Linux. Since you're using JDK 17 I'mma try that real quick; I upgraded straight to JDK 18 when it told me I was out of date.

Honestly, nearly every piece of technology I've ever interacted with has broken in some new, never before seen, way so this is par for my fucking course, (the computer science courses I took in college were a fucking nightmare, not because of the assignments but because I'm fucking cursed let me tell ya). I'm very unsurprised that you're having trouble replicating the issue.

**Reply by HydraheadHunter** [reply]
It currently won't attempt to compile on JDK 17 because its looking for JDK 18 even after a reboot and fresh download.

Edit: After a quick tutorial on how to set up java_home, the compile's now starting again, it also got passed the previous dead% of 75 and is now sitting on 81%, so we'll see how that goes...

**Reply by maruohon** [reply]
You probably need to either remove the Java 18 installation, or update the `JAVA_HOME` environment variable (or the `PATH` variable?). I'm not really sure how those environment variables work on Windows...

I'm not sure what the Java version support status is in Gradle and in Loom, so I'm not sure if building using Java 18 is supposed to work, at least using that Loom version (0.11.x).

But anyway, I don't know if you want to continue trying to build the mod yourself just to see what the issue was... but I also now built a custom version with the potion HUD shift disabled, here: https://masa.dy.fi/tmp/minecraft/mods/misc_test_versions/malilib-fabric-1.18.2-0.12.1-nopotionhudshift.1.jar

I didn't actually test this at all though, so let me know if it's not working I guess...

**Reply by HydraheadHunter** [reply]
I appreciate you so much. It is indeed working, thank you so much for helping me out, and sorry for the trouble. Have a lovely day.
